packages/playground/src/util/cacheHandlers.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  cloneElementInCode,
+  deleteElementInCode,
+  findElementInAST,
+} from './codeHandlers';
+
+const getDataFromCache = async (
```

Comment:
Do not combine cache open & AST logic in one helper function.
Reasons:
- there is no proper naming for such function
- you need to update the cache 
- you do not want to mess the code with the cache logic (it is a cross cutting concern - if we want to replace cache with something else it will not be in one proper place)

Generaly you want something like this

caches.open(....).then {
  read cache
  findAST//no need to know about the cache
  clone/delete
  write cache//this is important
}